scripts/codemods/typed_flags.py

```diff
@@ -0,0 +1,191 @@
+                # find the first function definition and insert it before there
+                for pos, b in enumerate(body):  # noqa: B007
+                    if m.matches(b, m.FunctionDef()):
+                        break
+                else:
+                    # so there wasn't any function, not sure why
+                    # doesn't really matter, we just insert at the end
+                    pos = len(body)
+                body.insert(pos, if_block)
+                node = node.with_deep_changes(node.body, body=body)
+
+            # find the init
+            for b in if_block.body.body:
+                if m.matches(b, m.FunctionDef(m.Name("__init__"))) and isinstance(
+                    b, cst.FunctionDef
+                ):
+                    init = b
+                    break
+            else:
+                raise RuntimeError
+            # add to the init the generated decorator
+            old_init = init
+            init = old_init.with_changes(decorators=[cst.Decorator(cst.Name("_generated"))])
+            node = node.deep_replace(old_init, init)  # type: ignore
+            node = node.with_deep_changes(init.params, kwonly_params=kwonly_params, star_kwarg=None)
+
+        else:
+            # the init exists, so we need to make overloads
+            # here's how:
+            # first, we use the init we've already found
+            # now, we need to insert two overloaded functions before it
+            # this can be done by getting an init `with_changes` and remove the body
+            # then we add the _generated decorator
+            # then we remove all parameters except for self, and save that as our first overload
+            # next, we need to add all of the flag values and that is our second overload
+            # then we put them together and insert them before the existing overload
+            # todo: this is very complicated, and maybe this visitor should be split into another visitor just to do this logic
+            assert init  # noqa: S101
+            no_body_init = init.with_changes(
+                body=cst.IndentedBlock([cst.SimpleStatementLine([cst.Expr(cst.Ellipsis())])]),
+                decorators=[
+                    cst.Decorator(cst.Name("overload")),
+                    cst.Decorator(cst.Name("_generated")),
+                ],
+            )
+
+            empty_init = no_body_init.with_changes(
+                params=cst.Parameters(
+                    params=[cst.Param(cst.Name("self"), cst.Annotation(cst.Name("NoReturn")))]
+                )
+            )
+            full_init = no_body_init.with_deep_changes(
+                no_body_init.params, kwonly_params=kwonly_params, star_kwarg=None
+            )
+            for pos, b in enumerate(body):
+                if m.matches(b, m.FunctionDef(m.Name("__init__"))):
+                    body.insert(pos, empty_init)
+                    body.insert(pos, full_init)
+                    break
```

Comment:
I believe this could use `body.index(init)` instead of looping?

Reply:
totally can.

Reply:
https://github.com/DisnakeDev/disnake/pull/778/commits/1338f5937c527e33fc61ae1fdcf86df4c6a7f97d

Reply:
https://github.com/DisnakeDev/disnake/pull/778/commits/621fdf0ea61e5f39d69f4b0bd6625171de696ca7